### PR TITLE
[MISC] Deprecate seqan3::views::persist

### PR DESCRIPTION
**Pull request by eseiler**
Resolves seqan/product_backlog#329

### CODE REVIEW

File: test/performance/search/search_benchmark.cpp [hunk 1]
```diff
@@ -142,7 +142,7 @@ std::vector<alphabet_t> generate_repeating_sequence(size_t const template_length
     len = (len + simulated_errors  > template_length) ? template_length - simulated_errors : len;
 
     return generate_reads(seq_template, repeats, len, simulated_errors, 0.15, 0.15)
-         | seqan3::views::persist
+         | seqan3::detail::persist
```
Comment (Irallia) [inline test/performance/search/search_benchmark.cpp:145]: There is no include for `seqan3::detail::persist`, do you want to add it?
#include <seqan3/core/detail/persist_view.hpp>

File: include/seqan3/search/fm_index/bi_fm_index.hpp [hunk 2]
```diff
@@ -16,8 +16,8 @@
 #include <seqan3/std/ranges>
 #include <utility>
 
+#include <seqan3/core/detail/persist_view.hpp>
```
Comment (Irallia) [inline include/seqan3/search/fm_index/bi_fm_index.hpp:19]: Do we need the include here, as it is not used in this file?

**Review by Irallia — CHANGES_REQUESTED** [review]
LGFM, just one include maybe missing and one is maybe not used.

**Review by Irallia — APPROVED** [review]
Thank you!

File: test/snippet/io/sam_file/sam_file_output_io_pipeline.cpp [hunk 3]
```diff
@@ -15,9 +14,8 @@ r001	147	ref	237	30	*	=	7	-39	CAGCGGCAT	*	NM:i:1
...
 {
 #if !SEQAN3_WORKAROUND_GCC_93983
-    seqan3::sam_file_input{std::istringstream{sam_file_raw}, seqan3::format_sam{}}
-        | seqan3::views::persist
-        | std::views::take(3) // take only the first 3 records
-        | seqan3::sam_file_output{std::ostringstream{}, seqan3::format_sam{}};
+    auto input_file = seqan3::sam_file_input{std::istringstream{sam_file_raw}, seqan3::format_sam{}};
+    input_file | std::views::take(3) // take only the first 3 records
+               | seqan3::sam_file_output{std::ostringstream{}, seqan3::format_sam{}};
 #endif // !SEQAN3_WORKAROUND_GCC_93983
```
Comment (marehr) [inline test/snippet/io/sam_file/sam_file_output_io_pipeline.cpp:20]: You did that for the other snippet, maybe here too?

**Review by marehr — APPROVED** [review]

**Pull request merged**